src/utils/FlowUtils.ts

```diff
@@ -31,61 +32,47 @@ export const V3_DEPOSIT_COMPARISON_KEYS = [
 
 export function filledSameDeposit(fillA: Fill, fillB: Fill): boolean {
   if (isV2Fill(fillA) && isV2Fill(fillB)) {
-    return (
-      fillA.depositId === fillB.depositId &&
-      fillA.originChainId === fillB.originChainId &&
-      fillA.amount.eq(fillB.amount) &&
-      fillA.destinationChainId === fillB.destinationChainId &&
-      fillA.relayerFeePct.eq(fillB.relayerFeePct) &&
-      fillA.recipient === fillB.recipient &&
-      fillA.depositor === fillB.depositor &&
-      fillA.message === fillB.message
-    );
+    return getV2RelayHash(fillA) === getV2RelayHash(fillB);
   } else if (isV3Fill(fillA) && isV3Fill(fillB)) {
-    return (
-      fillA.depositId === fillB.depositId &&
-      fillA.originChainId === fillB.originChainId &&
-      fillA.destinationChainId === fillB.destinationChainId &&
-      fillA.recipient === fillB.recipient &&
-      fillA.depositor === fillB.depositor &&
-      fillA.inputToken === fillB.inputToken &&
-      fillA.outputToken === fillB.outputToken &&
-      fillA.message === fillB.message &&
-      fillA.inputAmount.eq(fillB.inputAmount) &&
-      fillA.outputAmount.eq(fillB.outputAmount) &&
-      fillA.fillDeadline === fillB.fillDeadline &&
-      fillA.exclusivityDeadline === fillB.exclusivityDeadline &&
-      fillA.exclusiveRelayer === fillB.exclusiveRelayer
-    );
+    const { destinationChainId: chainA } = fillA;
+    const { destinationChainId: chainB } = fillB;
+    return chainA === chainB && getV3RelayHash(fillA, chainA) === getV3RelayHash(fillB, chainB);
```

Comment:
Is `chainA === chainB` neccessary since its part of the hash?

Reply:
Removed: https://github.com/across-protocol/sdk-v2/pull/549/commits/b16bd440f5cf31e9bed28488ac8630ddcb2c3bd7

We can also remove this function entirely in the not-too-distant future :raised_hands: